src/main/java/com/github/spotbugs/SpotBugsPlugin.java

```diff
@@ -245,15 +245,21 @@ protected void configureForSourceSet(final SourceSet sourceSet, SpotBugsTask tas
         taskMapping.map("classes", new Callable<FileCollection>() {
             @Override
             public FileCollection call() {
-                ConfigurableFileTree fileTree = StreamSupport.stream(sourceSet.getOutput().getClassesDirs().spliterator(), false)
-                    .map(project::fileTree)
-                    .reduce((lhs, rhs) -> {
-                        lhs.plus(rhs);
-                        return lhs;
-                    }).orElseThrow(() ->
-                        new InvalidUserDataException("No classes dir configured for source set " + sourceSet.getName())
-                    );
-                return fileTree.builtBy(sourceSet.getClassesTaskName());
+                /*
+                 * As a result of the changes made in gradle 4.0.
+                 * See https://docs.gradle.org/4.0/release-notes.html - Location of classes in the build directory
+                 * Compile no longer bundles all classes in one directory build-gradle/classes/main
+                 * but instead separates classes into build-gradle/classes/{language}/main.
+                 *
+                 * We must therefor retrieve all output directories. Filter away the once that don't exist. Add each
+                 * existing file tree dependency to specified task. And then return the complete fileCollection, contain
+                 * all .class files available for analysis.
+                 */
+                FileCollection presentClassDirs = sourceSet.getOutput().getClassesDirs().filter(File::exists);
```

Comment:
filtering prevents a NoSuchFileException from being throw down in fb/sb internals as it will attempt to access directories even though they don't exist. 
A better approach would be to handle this exception where it arises and skip this "potentially" costly operation here.